omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -46,6 +48,33 @@ public UiSessionContext() {
 
     }
 
+    public String getViewProvidersPrivilege()
+    {
+    	PrivilegeConstants privilegeConstants = new PrivilegeConstants();
+    	String viewProvidersPrivilege = "";
+    	try {
+    		// openmrsVersion 1.9
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("VIEW_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	try {
+    		// openmrsVersion 2.0
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("GET_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	return viewProvidersPrivilege;
+    }
+    
```

Comment:
I think I see where you're coming from and why you wrote this...It sucks that java doesn't have pre-processor directives like C/C++ do...this would be mega handy. There may not be another way short of doing separate modules which is quite insane. 

My objections are hereby removed.